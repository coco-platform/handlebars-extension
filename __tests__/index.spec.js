/**
 * @description - handlebars extensions
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// package
const path = require('path');
const fs = require('fs');
const hbs = require('handlebars');
// internal
const { inline, link } = require('../');
// scope
const options = { encoding: 'utf8' };

describe('@coco-platform/handlebars-extension', () => {
  beforeAll(() => {
    hbs.registerHelper('inline', inline);
    hbs.registerHelper('link', link);
  });

  it('should support inline expression', () => {
    const demo = path.resolve(__dirname, '__fixture__', 'inline.hbs');
    const template = fs.readFileSync(demo, options);
    const context = {
      criticals: [
        {
          type: 'stylesheet',
          content:
            '.CloseIcon-icon-2xww{transition:opacity .3s ease-out}.CloseIcon-icon-2xww:hover{opacity:.8}',
        },
        {
          type: 'script',
          content: `(function($) {$(document).ready(() => {});})(jQuery);`,
        },
        {},
      ],
    };
    const result = hbs.compile(template)(context);

    expect(result).toMatchSnapshot();
  });

  it('should support link expression', () => {
    const demo = path.resolve(__dirname, '__fixture__', 'link.hbs');
    const template = fs.readFileSync(demo, options);
    const context = {
      resources: [
        'https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.css',
        {
          href:
            'https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.css',
        },
        {
          rel: 'dns-prefetch',
          href: `//static.zhimg.com`,
        },
        'https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.js',
        {
          src:
            'https://static.zhihu.com/heifetz/main.signflow.6af7025179e6b1979aca.js',
          crossorigin: true,
        },
        {},
      ],
    };
    const result = hbs.compile(template)(context);

    expect(result).toMatchSnapshot();
  });

  it('should return empty with un-supported resource', () => {
    expect(inline(200)).toEqual('');
    expect(link(200)).toEqual('');
  });
});
