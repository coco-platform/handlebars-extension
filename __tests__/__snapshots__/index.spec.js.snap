// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@coco-platform/handlebars-extension should support inline expression 1`] = `
"  <style type=\\"text/css\\">
.CloseIcon-icon-2xww{transition:opacity .3s ease-out}.CloseIcon-icon-2xww:hover{opacity:.8}
</style>
  <script type=\\"application/javascript\\">
(function($) {$(document).ready(() &#x3D;&gt; {});})(jQuery);
</script>
  
"
`;

exports[`@coco-platform/handlebars-extension should support link expression 1`] = `
"  <link rel=\\"stylesheet\\" href=\\"https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.css\\">
  <link rel=\\"stylesheet\\" href=\\"https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.css\\">
  <link rel=\\"dns-prefetch\\" href=\\"//static.zhimg.com\\">
  <script type=\\"application/javascript\\" src=\\"https://static.zhihu.com/heifetz/main.app.c994694b7b8c848b345c.js\\"></script>
  <script type=\\"application/javascript\\" src=\\"https://static.zhihu.com/heifetz/main.signflow.6af7025179e6b1979aca.js\\" crossorigin=\\"anonymous\\"></script>
  
"
`;
